litani

```diff
@@ -95,220 +63,19 @@ def get_args():
-            "help": "also print out capabilities with their descriptions",
-            "action": "store_true"
-    }]:
-        flags = arg.pop("flags")
-        caps_pars.add_argument(*flags, **arg)
-
-    run_build_pars = subs.add_parser("run-build")
-    run_build_pars.set_defaults(func=run_build)
-    for arg in [{
-            "flags": ["-n", "--dry-run"],
-            "help": "don't actually run jobs, just act like they succeeded",
-            "action": "store_true",
-    }, {
-            "flags": ["-j", "--parallel"],
-            "metavar": "N",
-            "help": "run N jobs in parallel. 0 means infinity, default is "
-                    "based on the number of cores on this system.",
-    }, {
-            "flags": ["-o", "--out-file"],
-            "metavar": "F",
-            "help": "periodically write JSON representation of the run"
-    }, {
-            "flags": ["--fail-on-pipeline-failure"],
-            "action": "store_true",
-            "help": "exit with a non-zero return code if some pipelines failed"
-    }, {
-            "flags": ["--no-pipeline-dep-graph"],
-            "action": "store_true",
-            "help": "do not attempt to generate pipeline dependency graph"
-    }]:
-        flags = arg.pop("flags")
-        run_build_pars.add_argument(*flags, **arg)
-    mutex = run_build_pars.add_mutually_exclusive_group()
-    for arg in [{
-            "flags": ["-p", "--pipelines"],
-            "metavar": "P",
-            "nargs": "+",
-            "help": "only run jobs that are in the specified pipelines"
-    }, {
-            "flags": ["-s", "--ci-stage"],
-            "metavar": "S",
-            "help": (
-                "only run jobs that are part of the specified ci stage. S "
-            )
-    }]:
-        flags = arg.pop("flags")
-        mutex.add_argument(*flags, **arg)
-
-    add_job_pars = subs.add_parser("add-job")
-    add_job_pars.set_defaults(func=add_job)
-    for group_name, args in lib.jobs.get_add_job_args():
-        group = add_job_pars.add_argument_group(title=group_name)
-        for arg in args:
-            flags = arg.pop("flags")
-            group.add_argument(*flags, **arg)
-
-    exec_job_pars = subs.add_parser("exec")
-    exec_job_pars.set_defaults(func=exec_job)
-    for group_name, args in get_exec_job_args():
-        group = exec_job_pars.add_argument_group(title=group_name)
-        for arg in args:
-            flags = arg.pop("flags")
-            group.add_argument(*flags, **arg)
+    lib.init.add_subparser(subs)
+    lib.jobs.add_add_jobs_subparser(subs)
+    lib.jobs.add_get_jobs_subparser(subs)
+    lib.jobs.add_set_jobs_subparser(subs)
+    lib.jobs.add_transform_jobs_subparser(subs)
+    lib.graph.add_subparser(subs)
+    lib.run_build.add_subparser(subs)
+    lib.exec.add_subparser(subs)
+    lib.capabilities.add_subparser(subs)
+    lib.run_printer.add_subparser(subs)
+    lib.litani_report.add_print_html_dir_subparser(subs)
```

Comment:
Suggest:
* the entry points are all in a directory called `entry_points`
* the code here iterates over all the modules and calls `add_subparser`

Reply:
From our offline conversation: Its not clear what the `entry_points` directory would look like. It could have a file for each individual subcommand, some with multiple subcommands, or a single big file, which may kind of defeat the purpose of the refactoring. What I've done instead was to iterate over the `lib` directory and call `add_subparser` on each file which contains that method